### PR TITLE
docs: typescript - provide information about exactOptionalPropertyTypes

### PR DESCRIPTION
**Pull request by chrisk-7777**
## Related Issues or Discussions

Fixes #1629

## Summary

Add more information about `exactOptionalPropertyTypes`

## Check List

- [X] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/guides/initialize-state-with-props.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ const createBearStore = (initProps?: Partial<BearProps>) => {
 }
 ```
 
+Please see the [TypeScript Guide](../guides/typescript.md#store-creator-with-createstore)
```
Comment (sewera) [inline docs/guides/initialize-state-with-props.md:35]: I think going up the dir and back into it is redundant.

```suggestion
Please see the [TypeScript Guide](./typescript.md#store-creator-with-createstore)
```

File: docs/guides/typescript.md [hunk 2]
```diff
@@ -460,6 +460,65 @@ type ExtractState<S> = S extends { get: () => infer X } ? X : never
 const useBearStore = createBoundedUseStore(bearStore)
 ```
 
+### Store creator with createStore
+
+As shown in the
+[initialize state with props guide](../guides/initialize-state-with-props.md##store-creator-with-createstore),
```
Comment (sewera) [inline docs/guides/typescript.md:466]:
```suggestion
[initialize state with props guide](./initialize-state-with-props.md##store-creator-with-createstore),
```

**Review by sewera — APPROVED** [review]
Apart from the small redundancy, I think it reads well, so LGTM.

**Comment by dai-shi** [comment]
@chrisk-7777 Please check prettier or lint.

@devanshj Would like to have a quick look?

**Comment by devanshj** [comment]
I don't have the bandwidth for a detailed review but in short...

- I don't think the change in typescript md is needed, the store initialisation md is already written in typescript, so maybe just add the note about `exactOptionalPropertyTypes` there. Other mds are written in js hence they have a corresponding recipe in typescript md, here that's not the case.

- `exactOptionalPropertyTypes` is anyway recommended, so that it prevents you from writing stuff like `set({ bears: undefined })`, so perhaps that can be added in the beginning of the typescript md in a different PR.

Just my two cents.

**Comment by chrisk-7777** [comment]
Yeah agreed, I was unsure about the duplication to be honest. I'm happy to just close this PR and replace it with just a note as you said

**Review by sewera — APPROVED** [review]

**Comment by sewera** [comment]
@dai-shi @chrisk-7777 

> Please check prettier or lint.

From what I've seen in logs, the file reported by Prettier is `src/middleware/devtools.ts`, but it wasn't changed in this PR.

**Comment by ifeelBALANCED** [comment]
Hi @dai-shi, @sewera, and everyone contributing. I have a question: do you mind if we transform the current Zustand repository to an NX/Turborepo package-based repository? The idea behind this is that it's cleaner to maintain and build all packages via these tools. It will also be better for providing support for Zustand in non-React libraries in the future.

**Comment by chrisk-7777** [comment]
Hi @ifeelBALANCED 

I think that would be more suited to a new discussion topic rather than a comment here. It might get lost as a comment to this PR

Thanks

**Pull request closed without merging**